internal/datasource/block_datasource.go

```diff
@@ -0,0 +1,9 @@
+package datasource
```

Comment:
I guess this is supposed to go the synchronizer package

Reply:
Done